### PR TITLE
BREAKING(model): get rid of `Model.example_output` in favor of `num_frames`, `receptive_field`, and `dimension` properties

### PR DESCRIPTION
**Pull request by hbredin**

### CODE REVIEW

**Comment by hbredin** [comment]
@clement-pages you might want to try this in your tutorial PR (regarding sending model to device prior training).

File: pyannote/audio/core/model.py [hunk 1]
```diff
@@ -270,9 +234,6 @@ def setup(self, stage=None):
             # setup custom validation metrics
             self.task.setup_validation_metric()
 
-            # cache for later (and to avoid later CUDA error with multiprocessing)
-            _ = self.example_output
-
```
Comment (hbredin) [inline pyannote/audio/core/model.py:275]: @clement-pages at least this line should no longer be a problem.

Comment (clement-pages) [reply pyannote/audio/core/model.py:275]: I'll test with these changes and let you know.

Comment (clement-pages) [reply pyannote/audio/core/model.py:275]: The problem of sending the model to the GPU is fixed, I update my tutorial PR to remove manual  sending.

**Comment by hbredin** [comment]
Manually checking that this (at least) does not break pipeline inference.

Benchmark | Before | After
----------|--------|-------
AISHELL | 12.27 | 12.26
AISHELLMix | 12.01 | 12.00
AliMeeting | 24.57 | WIP
AMI-SDM | 22.87 | 22.87
AMI | 18.81 |  18.81
AVA-AVD | 49.47 | 49.52
CALLHOME | 28.54 | 28.57
DIHARD | 21.46 | WIP
DIHARDMini | 21.47 | 21.49
Ego4D | 51.36 | 51.36
MSDWILD | 25.43 | 25.46
RAMC | 22.19 | WIP
REPERE | 7.94 | WIP
RevDotCom | 9.36 | WIP
ThisAmericanLife | 25.29 |
VoxConverse | 11.23 | WIP

**Comment by hbredin** [comment]
Failing on `AliMeeting` with 

```
Processing AliMeeting.SpeakerDiarization.Benchmark (test):   0%|          | 0/20 [00:01<?, ?file/s]
Traceback (most recent call last):
  File "[REDACTED]/src/pyannote-audio/pyannote/audio/core/pipeline.py", line 325, in __call__
    return self.apply(file, **kwargs)
  File "[REDACTED]/src/pyannote-audio/pyannote/audio/pipelines/speaker_diarization.py", line 478, in apply
    segmentations = self.get_segmentations(file, hook=hook)
  File "[REDACTED]/src/pyannote-audio/pyannote/audio/pipelines/speaker_diarization.py", line 224, in get_segmentations
    segmentations: SlidingWindowFeature = self._segmentation(file, hook=hook)
  File "[REDACTED]/src/pyannote-audio/pyannote/audio/core/inference.py", line 421, in __call__
    return self.slide(waveform, sample_rate, hook=hook)
  File "[REDACTED]/src/pyannote-audio/pyannote/audio/core/inference.py", line 313, in slide
    batch_outputs: Union[np.ndarray, Tuple[np.ndarray]] = self.infer(batch)
  File "[REDACTED]/src/pyannote-audio/pyannote/audio/core/inference.py", line 223, in infer
    raise exception
  File "[REDACTED]/src/pyannote-audio/pyannote/audio/core/inference.py", line 215, in infer
    outputs = self.model(chunks.to(self.device))
  File "[REDACTED]/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "[REDACTED]/src/pyannote-audio/pyannote/audio/models/segmentation/PyanNet.py", line 208, in forward
    outputs = self.sincnet(waveforms)
  File "[REDACTED]/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "[REDACTED]/src/pyannote-audio/pyannote/audio/models/blocks/sincnet.py", line 182, in forward
    outputs = F.leaky_relu(norm1d(pool1d(outputs)))
  File "[REDACTED]/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "[REDACTED]/lib/python3.10/site-packages/torch/nn/modules/pooling.py", line 92, in forward
    return F.max_pool1d(input, self.kernel_size, self.stride,
  File "[REDACTED]/lib/python3.10/site-packages/torch/_jit_internal.py", line 484, in fn
    return if_false(*args, **kwargs)
  File "[REDACTED]/lib/python3.10/site-packages/torch/nn/functional.py", line 696, in _max_pool1d
    return torch.max_pool1d(input, kernel_size, stride, padding, dilation, ceil_mode)
RuntimeError: max_pool1d: Expected 2D or 3D (batch mode) tensor with optional 0 dim batch size for input, but got:[32, 0, 80, 15975]
```

**Comment by hbredin** [comment]
Seems to be a problem with Waveform preprocessor.
Nothing related to this PR.
Merging but will need to have a look at this.

**Pull request merged**